src/pages/index.js

```diff
@@ -29,50 +29,165 @@ import {
   previewFooter,
   formValidationConfig,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 
-// New Card
+// =============================================================================
+// API + userInfo
+// =============================================================================
 
-function createCard({ name, link }) {
+let userID;
+let cardList;
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "a9aaf24d-57aa-44df-b0d5-364cf1f03985",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  professionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+Promise.all([api.getUserInfo(), api.getInitialCards()]).then(
+  ([user, initialCards]) => {
+    userInfo.setUserInfo(user);
+    userID = user._id;
+    userInfo.setUserInfo(user);
+
+    cardList = new Section(
+      {
+        items: initialCards,
+        renderer: ({ name, link, isLiked, _id, userId, ownerId }) => {
+          const newCard = createCard({
+            name,
+            link,
+            isLiked,
+            _id,
+            userId,
+            ownerId,
+          });
+          cardList.addItem(newCard);
+        },
+      },
+      ".cards__list"
+    );
+
+    cardList.renderItems();
+  }
+);
+// =============================================================================
+// Handle Clicks
+// =============================================================================
+
+function handleLikeClick() {
+  if (this._likes) {
+    api
+      .unlikeCard(this._id)
+      .then((res) => {
+        this.setLikes(res.isLiked);
+      })
+      .catch((error) => {
+        console.log(`An error has occured ${error}`);
+      });
+  } else {
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards